### PR TITLE
Support for reading uints in a similar manner to floats

### PR DESCRIPTION
**Pull request by BobbyAnguelov**

### CODE REVIEW

**Comment by jkuhlmann** [comment]
Looks fine to me. 👍 
@prideout What do you think?

**Comment by prideout** [comment]
Oh, I'm guessing the use case is skinning joints, which are vec4 so read_index wouldn't work.

In my case, I only implement skinning on the GPU so there's no reason to parse them on the CPU.

Anyway: again, I think this feature is fine if we add to the doc block and to the test.

**Comment by prideout** [comment]
We also don't need the `normalized` argument in the new function.

Before merging, let's confirm that this feature is really needed.

**Comment by BobbyAnguelov** [comment]
Yes, it is exactly for that use case. I parse joints on the cpu side since I'm using gltf as import format not a runtime one. It just feel really strange to call read float for non float data and then cast it back. Additionally depending how the data is read you could be flirting with UB. This just felt simple and more obvious for a user.

As a first time user of the lib, it very unclear when I should call what function. It took me a while to try to figure out how to correctly read the joint indices, and I couldn't so I hand read them from the accessor. It would have never occurred to me to call read float to read non-float data. 

An argument could also be made as to why read index exists? Can't I just call read float? I'm going to assume I can read index data through the read float function as well. Maybe the new function can be renamed to read_indices instead of read_uint?

**Comment by prideout** [comment]
Triangle indices can be large so their values cannot always be represented with fp32, which is why the existing `read_index` function exists.  (joint indices tend to be smaller than triangle indices)

Anyway, I agree that the back-and-forth with float is sort of awkward, moreover this could be useful for future glTF features or extensions, so let's move forward with this.

I think the `cgltf_accessor_read_uint` name is good.  Only thing remaining is adding to the conversion test and adding to the reference section at the top of the file.

**Comment by BobbyAnguelov** [comment]
I made the changes as request, though I'm not entirely sure I understand the purpose of the conversion test. Can you please verify that I haven't done something idiotic there?

File: cgltf.h [hunk 1]
```diff
@@ -71,6 +71,11 @@
  * floating point, assuming that `cgltf_load_buffers` has already been called. The passed-in element
  * size is the number of floats in the output buffer, which should be in the range [1, 16]. Returns
  * false if the passed-in element_size is too small, or if the accessor is sparse.
+
```
Comment (prideout) [inline cgltf.h]: Remove empty line

File: cgltf.h [hunk 2]
```diff
@@ -1787,6 +1794,77 @@ cgltf_size cgltf_accessor_unpack_floats(const cgltf_accessor* accessor, cgltf_fl
 	return element_count * floats_per_element;
 }
 
+static cgltf_uint cgltf_component_read_uint(const void* in, cgltf_component_type component_type)
+{
+	switch (component_type)
+	{
+		case cgltf_component_type_r_8:
+		return *((const int8_t*) in);
```
Comment (prideout) [inline cgltf.h]: The return statements are missing an indention level.

File: test/test_conversion.cpp [hunk 3]
```diff
@@ -51,16 +52,32 @@ int main(int argc, char** argv)
...
 				return -1;
 			}
 		}
+		if (blob->has_max && blob->has_min && blob->component_type != cgltf_component_type_r_32f && blob->component_type != cgltf_component_type_invalid)
+		{
+			cgltf_uint min0 = std::numeric_limits<unsigned int>::max();
+			cgltf_uint max0 = std::numeric_limits<unsigned int>::lowest();
+			for ( cgltf_size index = 0; index < blob->count; index++ )
+			{
+				cgltf_accessor_read_uint( blob, index, element_uint, 4 );
```
Comment (prideout) [inline test/test_conversion.cpp]: Looks fine, assuming at least one of the test assets actually causes this codeline to be reached.  Just some sort of basic sanity test was all I was looking for.

File: cgltf.h (hunk 1 above)
Comment (prideout) [reply cgltf.h]: (or add a `*` at the beginning)

**Comment by BobbyAnguelov** [comment]
Hi, is there anything I need to do further to get this merged into the branch?

**Comment by prideout** [comment]
Looks good to me, it's up to Johannes now.

**Comment by jkuhlmann** [comment]
Thank you both for the work on this!  😀

**Pull request merged**